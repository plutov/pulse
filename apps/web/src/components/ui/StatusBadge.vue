<template>
  <q-badge :color="getBadgeColor(value)" :label="value" />
</template>

<script setup lang="ts">
defineProps<Props>();

type Status =
  | "active"
  | "success"
  | "failure"
  | "timeout"
  | "paused"
  | undefined;

interface Props {
  value: Status;
}

const getBadgeColor = (status: Status): string => {
  switch (status) {
    case "active":
    case "success":
      return "positive";
    case "failure":
      return "negative";
    case "timeout":
      return "warning";
    case "paused":
      return "orange";
    default:
      return "grey";
  }
};
</script>
