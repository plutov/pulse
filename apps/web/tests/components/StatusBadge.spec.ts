import { mount } from "@vue/test-utils";
import { describe, it, expect } from "vitest";
import StatusBadge from "../../src/components/ui/StatusBadge.vue";

describe("StatusBadge.vue", () => {
  it("renders the correct message", () => {
    const wrapper = mount(StatusBadge, { props: { value: "success" } });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
