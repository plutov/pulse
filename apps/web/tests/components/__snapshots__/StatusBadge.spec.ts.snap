// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StatusBadge.vue > renders the correct message 1`] = `"<q-badge outline="" color="positive" label="success"></q-badge>"`;
